Why?!

cc @isuruf @erykoff 

Two errors

1. not building the static libgfortran.a is causing an error in the tests locally when linking

env:

```
2020-10-07T18:47:35.7880620Z     cctools_osx-arm64:           949.0.1-h5970944_14           conda-forge/label/llvm_rc
2020-10-07T18:47:35.7881590Z     clang:                       11.0.0.rc1-default_hf57f61e_0 conda-forge/label/llvm_rc
2020-10-07T18:47:35.7882520Z     clang_osx-arm64:             11.0.0.rc1-hd819f65_7         conda-forge/label/llvm_rc
2020-10-07T18:47:35.7883420Z     clangxx:                     11.0.0.rc1-default_hf57f61e_0 conda-forge/label/llvm_rc
2020-10-07T18:47:35.7884470Z     compiler-rt:                 11.0.0.rc1-hb5e1fa0_2         conda-forge/label/llvm_rc
2020-10-07T18:47:35.7885600Z     compiler-rt_osx-64:          11.0.0.rc1-h62eb4f5_2         conda-forge/label/llvm_rc
2020-10-07T18:47:35.7886660Z     gfortran_impl_osx-arm64:     11.0.0.dev0-h6649a6a_12       conda-forge              
2020-10-07T18:47:35.7887690Z     gfortran_osx-arm64:          11.0.0.dev0-hda2b30b_10       local                    
2020-10-07T18:47:35.7888720Z     gmp:                         6.2.0-hb1e8313_2              conda-forge              
2020-10-07T18:47:35.7889850Z     isl:                         0.22.1-h3a6d616_1             conda-forge              
2020-10-07T18:47:35.7890880Z     ld64_osx-arm64:              530-h8c8ad8c_14               conda-forge/label/llvm_rc
2020-10-07T18:47:35.7891920Z     ldid:                        2.1.2-h79c402e_1              conda-forge              
2020-10-07T18:47:35.7892950Z     libclang-cpp11:              11.0.0.rc1-default_hf57f61e_0 conda-forge/label/llvm_rc
2020-10-07T18:47:35.7894000Z     libcxx:                      11.0.0.rc1-h439d374_2         conda-forge/label/llvm_rc
2020-10-07T18:47:35.7895470Z     libgfortran-devel_osx-arm64: 11.0.0.dev0-h181927c_12       conda-forge              
2020-10-07T18:47:35.7896520Z     libiconv:                    1.16-haf1e3a3_0               conda-forge              
2020-10-07T18:47:35.7897540Z     libllvm11:                   11.0.0.rc1-hf85e3d2_4         conda-forge/label/llvm_rc
2020-10-07T18:47:35.7898580Z     llvm-tools:                  11.0.0.rc1-hf85e3d2_4         conda-forge/label/llvm_rc
2020-10-07T18:47:35.7899610Z     mpc:                         1.1.0-ha57cd0f_1009           conda-forge              
2020-10-07T18:47:35.7900640Z     mpfr:                        4.0.2-h72d8aaf_1              conda-forge              
2020-10-07T18:47:35.7901660Z     tapi:                        1100.0.11-h879752b_0          conda-forge              
2020-10-07T18:47:35.7902700Z     zlib:                        1.2.11-h7795811_1009          conda-forge              
```

error:

```
2020-10-07T18:47:54.5759960Z Library search paths:
2020-10-07T18:47:54.5761380Z 	/Users/runner/miniforge3/conda-bld/gfortran_osx-arm64_1602096365103/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_place/lib/gcc/arm64-apple-darwin20.0.0/11.0.0
2020-10-07T18:47:54.5762740Z 	/Applications/Xcode_11.7.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.9.sdk/usr/lib
2020-10-07T18:47:54.5763300Z Framework search paths:
2020-10-07T18:47:54.5763860Z 	/Applications/Xcode_11.7.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.9.sdk/System/Library/Frameworks/
2020-10-07T18:47:54.5764830Z ld: library not found for -lgfortran
2020-10-07T18:47:54.5765240Z collect2: error: ld returned 1 exit status
```

2. we hit the stack protector thing again on gcc 9.3.0

env:

```
2020-10-07T18:45:13.0870124Z     _libgcc_mutex:               0.1-conda_forge               conda-forge              
2020-10-07T18:45:13.0870800Z     _openmp_mutex:               4.5-1_gnu                     conda-forge              
2020-10-07T18:45:13.0871452Z     binutils_impl_linux-64:      2.35-h18a2f87_9               conda-forge              
2020-10-07T18:45:13.0872126Z     ca-certificates:             2020.6.20-hecda079_0          conda-forge              
2020-10-07T18:45:13.0872770Z     cctools_osx-arm64:           949.0.1-h8d9e5e6_14           conda-forge/label/llvm_rc
2020-10-07T18:45:13.0873404Z     clang:                       11.0.0.rc1-default_hde54327_0 conda-forge/label/llvm_rc
2020-10-07T18:45:13.0874058Z     clang_osx-arm64:             11.0.0.rc1-hd819f65_7         conda-forge/label/llvm_rc
2020-10-07T18:45:13.0874697Z     clangxx:                     11.0.0.rc1-default_hde54327_0 conda-forge/label/llvm_rc
2020-10-07T18:45:13.0875350Z     compiler-rt:                 11.0.0.rc1-hc9558a2_2         conda-forge/label/llvm_rc
2020-10-07T18:45:13.0876035Z     compiler-rt_linux-64:        11.0.0.rc1-hfe2e4e2_2         conda-forge/label/llvm_rc
2020-10-07T18:45:13.0876751Z     compiler-rt_osx-64:          11.0.0.rc1-h62eb4f5_2         conda-forge/label/llvm_rc
2020-10-07T18:45:13.0877679Z     gcc_impl_linux-64:           9.3.0-h28f5a38_17             conda-forge              
2020-10-07T18:45:13.0878375Z     gfortran_impl_osx-arm64:     11.0.0.dev0-h7146c8e_12       conda-forge              
2020-10-07T18:45:13.0879014Z     gfortran_osx-arm64:          11.0.0.dev0-h0dbfe43_10       local                    
2020-10-07T18:45:13.0879668Z     gmp:                         6.2.0-he1b5a44_2              conda-forge              
2020-10-07T18:45:13.0880292Z     isl:                         0.22.1-he17e9d3_1             conda-forge              
2020-10-07T18:45:13.0881220Z     kernel-headers_linux-64:     2.6.32-h77966d4_13            conda-forge              
2020-10-07T18:45:13.0881877Z     ld64_osx-arm64:              530-had50491_14               conda-forge/label/llvm_rc
2020-10-07T18:45:13.0882546Z     ld_impl_linux-64:            2.35-h769bd43_9               conda-forge              
2020-10-07T18:45:13.0883175Z     ldid:                        2.1.2-h6eb9d5d_1              conda-forge              
2020-10-07T18:45:13.0883824Z     libclang-cpp11:              11.0.0.rc1-default_hde54327_0 conda-forge/label/llvm_rc
2020-10-07T18:45:13.0884637Z     libcxx:                      11.0.0.rc1-hc9558a2_2         conda-forge/label/llvm_rc
2020-10-07T18:45:13.0885296Z     libcxxabi:                   11.0.0.rc1-2                  conda-forge/label/llvm_rc
2020-10-07T18:45:13.0885941Z     libgcc-devel_linux-64:       9.3.0-hfd08b2a_17             conda-forge              
2020-10-07T18:45:13.0886580Z     libgcc-ng:                   9.3.0-h5dbcf3e_17             conda-forge              
2020-10-07T18:45:13.0887254Z     libgfortran-devel_osx-arm64: 11.0.0.dev0-h181927c_12       conda-forge              
2020-10-07T18:45:13.0887892Z     libgomp:                     9.3.0-h5dbcf3e_17             conda-forge              
2020-10-07T18:45:13.0888542Z     libiconv:                    1.16-h516909a_0               conda-forge              
2020-10-07T18:45:13.0889209Z     libllvm11:                   11.0.0.rc1-he513fc3_4         conda-forge/label/llvm_rc
2020-10-07T18:45:13.0889876Z     libstdcxx-ng:                9.3.0-h2ae2ef3_17             conda-forge              
2020-10-07T18:45:13.0890511Z     libuuid:                     2.32.1-h14c3975_1000          conda-forge              
2020-10-07T18:45:13.0891156Z     llvm-tools:                  11.0.0.rc1-he513fc3_4         conda-forge/label/llvm_rc
2020-10-07T18:45:13.0891787Z     mpc:                         1.1.0-h04dde30_1009           conda-forge              
2020-10-07T18:45:13.0892440Z     mpfr:                        4.0.2-he80fd80_1              conda-forge              
2020-10-07T18:45:13.0893069Z     openssl:                     1.1.1h-h516909a_0             conda-forge              
2020-10-07T18:45:13.0893723Z     sysroot_linux-64:            2.12-h77966d4_13              conda-forge              
2020-10-07T18:45:13.0894358Z     tapi:                        1100.0.11-hc9558a2_0          conda-forge              
2020-10-07T18:45:13.0895140Z     zlib:                        1.2.11-h516909a_1009          conda-forge              
```

error:

```
2020-10-07T18:45:53.0831302Z  "/home/conda/feedstock_root/build_artifacts/gfortran_osx-arm64_1602096207912/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeho/bin/clang-11" -cc1as -triple arm64-apple-macosx10.9.0 -filetype obj -main-file-name - -target-cpu apple-a7 -target-feature +fp-armv8 -target-feature +neon -target-feature +crypto -target-feature +zcm -target-feature +zcz -target-feature +sha2 -target-feature +aes -fdebug-compilation-dir /home/conda/feedstock_root/build_artifacts/gfortran_osx-arm64_1602096207912/test_tmp -dwarf-debug-producer "clang version 11.0.0 (https://github.com/conda-forge/clangdev-feedstock ddcabce346bf6afd60d6a1cc86dc5a105628b002)" -dwarf-version=2 -mrelocation-model pic -o hello -
2020-10-07T18:45:53.1126039Z <stdin>:25:15: error: index must be an integer in range [-256, 255].
2020-10-07T18:45:53.1126785Z         ldr     x2, [x2, ___stack_chk_guard];momd
2020-10-07T18:45:53.1127178Z                          ^
2020-10-07T18:45:53.1127846Z <stdin>:48:15: error: index must be an integer in range [-256, 255].
2020-10-07T18:45:53.1128322Z         ldr     x0, [x0, ___stack_chk_guard];momd
2020-10-07T18:45:53.1128653Z                          ^
```

--- a/recipe/build.sh
+++ b/recipe/build.sh
@@ -12,6 +12,7 @@ fi
 if [[ "$cross_target_platform" == "osx-arm64" ]]; then
   export FFLAGS="-march=armv8.3-a ${FFLAGS}"
   export DEBUG_FFLAGS="-march=armv8.3-a ${DEBUG_FFLAGS}"
+  export FFLAGS=${FFLAGS//"-fstack-protector"/"-fno-stack-protector"}
 fi
 
 # pushd ${PREFIX}/bin

--- a/recipe/build.sh
+++ b/recipe/build.sh
@@ -13,6 +13,7 @@ if [[ "$cross_target_platform" == "osx-arm64" ]]; then
   export FFLAGS="-march=armv8.3-a ${FFLAGS}"
   export DEBUG_FFLAGS="-march=armv8.3-a ${DEBUG_FFLAGS}"
   export FFLAGS=${FFLAGS//"-fstack-protector"/"-fno-stack-protector"}
+  export FFLAGS=${DEBUG_FFLAGS//"-fstack-protector"/"-fno-stack-protector"}
 fi
 
 # pushd ${PREFIX}/bin

--- a/recipe/run_test.sh
+++ b/recipe/run_test.sh
@@ -1,6 +1,6 @@
 #!/bin/bash
 
-if [[ "$target_platform" == osx-64 ]]; then
+if [[ "${target_platform}" == "osx-64" && "${cross_target_platform}" == "osx-64" ]]; then
   RUN_EXE=""
 else
   RUN_EXE="echo"

